Simplified/AxisService/NYPLAxisService.swift

```diff
@@ -72,253 +56,70 @@ import Foundation
-  
   // MARK: - Initialization
-  init(delegate: NYPLBookDownloadBroadcasting?,
-       isbn: String,
-       bookVaultId: String,
-       fileURL: URL,
+  init(axisItemDownloader: NYPLAxisItemDownloading,
        book: NYPLBook,
-       deviceInfoProviding: NYPLDeviceInfoProviding,
-       packageEndpointproviding: NYPLAxisPackageEndpointProviding,
-       packagePathProviding: NYPLAxisPackagePathPrefixProviding,
-       axisKeysProviding: NYPLAxisKeysProviding,
-       assetWriter: NYPLAssetWriting,
-       cypher: NYPLRSACryptographing?,
        dedicatedWriteURL: URL,
-       downloader: NYPLAxisContentDownloading?) {
+       delegate: NYPLBookDownloadBroadcasting?,
+       licenseService: NYPLAxisLicenseHandling,
+       metadataDownloader: NYPLAxisMetadataContentHandling,
+       packageDownloader: NYPLAxisPackageHandling) {
     
-    self.axisKeysProviding = axisKeysProviding
-    self.isbn = isbn
-    self.fileURL = fileURL
-    self.delegate = delegate
-    self.bookVaultId = bookVaultId
-    self.baseURL = axisKeysProviding.baseURL.appendingPathComponent(isbn)
+    self.axisItemDownloader = axisItemDownloader
     self.book = book
-    self.deviceInfoProviding = deviceInfoProviding
-    self.packagePathProviding = NYPLAxisPackagePathPrefixProvider()
-    self.dispatchGroup = DispatchGroup()
-    self.assetWriter = assetWriter
-    self.packageEndpointproviding = packageEndpointproviding
-    self.cypher = cypher
     self.dedicatedWriteURL = dedicatedWriteURL
-    self.downloader = downloader
+    self.delegate = delegate
+    self.licenseService = licenseService
+    self.metadataDownloader = metadataDownloader
+    self.packageDownloader = packageDownloader
+    super.init()
+    weak var weakSelf = self
+    self.axisItemDownloader.delegate = weakSelf
   }
   
   /// Fulfill AxisNow license. Notifies NYPLBookDownloadBroadcasting upon completion or failure.
-  
   /// - Parameters:
   ///   - downloadTask: downloadTask download task
   @objc func fulfillAxisLicense(downloadTask: URLSessionDownloadTask) {
     DispatchQueue.global(qos: .utility).async {
-      self.downloadLicense(forBook: self.book)
-      self.downloadEncryption(forBook: self.book)
-      self.downloadContainer(forBook: self.book)
-      self.downloadPackage(forBook: self.book)
-      self.downloadAssetsFromPackage(forBook: self.book)
-      
-      self.dispatchGroup.notify(queue: .global(qos: .utility)) {
-        _ = self.delegate?.replaceBook(self.book,
-                                       withFileAtURL: self.dedicatedWriteURL,
-                                       forDownloadTask: downloadTask)
+      self.downloadAndValidateLicense()
+      self.downloadMetadataContent()
+      self.downloadPackage()
+      self.axisItemDownloader.dispatchGroup.notify(queue: .global(qos: .utility)) {
+        // weak self will result in deallocation before book completion
```

Comment:
which object would get dealloced? If it's `self` this likely means there's an ownership problem on the service.

Reply:
It's AxisService. The reason for that is the only other strong reference to AxisService exists inside NYPLMyBooksDownloadCenter::URLSession:downloadTask:didFinishLoadingToURL method because we need a different AxisService for each book download.

Reply:
do you mean the reference [created here](https://github.com/NYPL-Simplified/Simplified-iOS/blob/01749343a8eee678e1e711d5b91c0fc720688c0e/Simplified/MyBooks/NYPLMyBooksDownloadCenter.m#L352)? If so I see what you mean: the object created there is immediately released so there's no remaining strong reference beside the capture done by the closure here.

This works as long as this closure is kept alive: once the dispatch group releases the closure the service will also be dealloced and that's probably not what you want. If you didn't observe this it's likely because the run time wasn't aggressive in reclaiming memory. 

Either way though the ownership hierarchy here looks reversed to me: it shouldn't be a low level closure holding the controlling service alive.

Since this PR is already big, I'd prefer actually if you file a ticket to fix this separately rather than fixing it now.